packages/components/docs/QueryModel.md

```diff
@@ -0,0 +1,164 @@
+
+// Here we use the name ExampleComponentImpl, users will not use this component directly, only the wrapped version below
+// which we expose to users as ExampleComponent.
+class ExampleComponentImpl extends PureComponent<Props> {
+    render() {
+        const { actions, queryModels, title } = this.props;
+        const model = queryModels.model;
+
+        return (
+            <div className="example-component">
+                <div>{title}</div>
+                <div>
+                    {model.isLoading && <LoadingSpinner />}
+                    {!model.isLoading && <span>The model has {model.rowCount} rows</span>}
+                </div>
+            </div>
+        );
+    }
+}
+
+// Next wrap your component with withQueryModels, here we set the type to MyComponentProps so the returned component,
+// ExampleComponent, can be used in a type safe manner. In this case, if the user forgets to pass in a title we'll get
+// a compiler error as intended.
+export const ExampleComponent = withQueryModels<MyComponentProps>(ExampleComponentImpl);
+
+// The component returned from withQueryModels, ExampleComponent in this case, now has the following props type:
+// MyComponentProps & MakeQueryModels. To use the component you pass it a queryConfig and it will isntantiate the
+// models you want and pass them to ExampleComponentImpl:
+const queryConfigs = { model: { schemaQuery: SchemaQuery.create('assay.general.amino acids', 'Runs') }};
+<ExampleComponent title="My Example Component" queryConfigs={queryConfigs} autoLoad />
+```
+
+There are more examples of withQueryModels in our QueryModel StoryBook story found in `/src/stories/QueryModel.tsx`.
+
+To summarize:
+ - Create a component that implements the `InjectedQueryModels` interface
+ - Wrap your component with `withQueryModels` (e.g `const MyComponentWithModels = withQueryModels(MyComponent)`)
+ - Use your wrapped component: `<MyComponentWithModels queryConfigs={myQueryConfigs} />`
+
+### GridPanel
+
+One of the most common use case for QueryModel is rendering data into a grid. We provide an easy way to render a grid
+via the `GridPanel` component. `GridPanel` has many props, but the only props you need to pass are a `model` (which is
+a `QueryModel`) and `actions` which is an `Actions` object. If you've wrapped your component in `withQueryModels` you
+should already have these.
+
+To render a "fully featured" GridPanel you would do this:
+
+```tsx
+<GridPanel model={myModel} actions={actions} />
+```
+
+Which would render something like this:
+
+![](default_grid_panel.png)
+
+To render a "minimal" GridPanel, you could do this:
+
+```tsx
+<GridPanel
+    model={myModel}
+    actions={actions}
+    asPanel={false}
+    showOmniBox={false}
+    showButtonBar={false}
+    allowSelections={false}
+/>
+```
+
+Which would render something like this:
+
+![](minimal_grid_panel.png)
+
+We also provide an implementation called `GridPanelWithModel` that lets you pass in a `queryConfigs` prop so you don't
```

Comment:
this line says it lets you pass in `queryConfigs` (plural) but the example below is singular. does it let you do both?

Reply:
I missed that after I changed the API, good catch.